freqtrade/optimize/backtesting.py

```diff
@@ -682,14 +727,23 @@ def _enter_trade(self, pair: str, row: Tuple, direction: str,
                     is_open=True,
                     enter_tag=entry_tag,
                     exchange=self._exchange_name,
-                    is_short=(direction == 'short'),
+                    is_short=is_short,
                     trading_mode=self.trading_mode,
                     leverage=leverage,
-                    orders=[]
+                    interest_rate=interest_rate,
+                    isolated_liq=isolated_liq,
+                    orders=[],
                 )
 
             trade.adjust_stop_loss(trade.open_rate, self.strategy.stoploss, initial=True)
 
+            if self.trading_mode == TradingMode.FUTURES:
+                if isolated_liq is None:
+                    raise OperationalException(
+                        f'isolated_liq is none for {pair} while trading futures, '
+                        'this should never happen')
```

Comment:
@xmatthias You mentioned once that you only want to use `assert` in tests, but it might be good to use assert instead of all these conditional checks with `OperationException`s. [I asked mypy about it a while ago and that was their advice](https://github.com/python/mypy/issues/10785)

Reply:
Assert statements can be disabled on a interpreter level with global variables. [how to](https://realpython.com/python-assert-statement/#disabling-assertions-in-production-for-performance).

so when you use asserts - you also assume that it doesn't matter if they are disabled.

I'm not sure about doing these checks this way - you could also simplify this with `if not isolated_liq: set_liquidation()`.

we should be certain enough about our implementation of `_leverage_prep` to be sure the above never happens (Futures + no liquidation returned). 
If we are not - then the implementation needs to be improved.